### DISCUSSION TITLE
Model deployment right after training?

### DISCUSSION THREAD

**Discussion by james-tn** [microsoft/dstoolkit-mlops-base#67] [General]
I see that you have the model deployment pipelines directly follow training. This means right after training is done, the output will be deployed?
Consider a little bit more realistic scenario where CD is a separate pipeline and user can decide on how to trigger it
such as manually or after a model testing pipeline where the produced model from training is tested against a holdout dataset and compare with model in production or when a new version of training dataset is available.

**Comment by FlorianPydde** [comment]
Hi james. You're right. The purpose of this template is to accelerate the implementation of MLOps and, thus, shows the basic components and workflow of MLOps with azure. 

You are right in that the training and deployment ought to be separated but this is left to the teams to fine-tune it to their environments
